Keda.Cosmosdb.Scaler/src/Services/ExternalScalerService.cs

```diff
@@ -0,0 +1,175 @@
+                                        request.ScalerMetadata[CosmosDbTriggerMetadata.CollectionName])),
+                TargetSize = 1
+            });
+            return Task.FromResult(resp);
+        }
+
+        public override async Task<GetMetricsResponse> GetMetrics(GetMetricsRequest request, ServerCallContext context)
+        {
+            var resp = new GetMetricsResponse();
+            try
+            {
+                var trigger = CreateTriggerFromMetadata(request.ScaledObjectRef.ScalerMetadata);
+                long workToBeDone = await GetEstimatedWork(trigger);
+
+                var metricName = request.ScaledObjectRef.ScalerMetadata["MetricName"];
+                resp.MetricValues.Add(new MetricValue
+                {
+                    MetricName = metricName,
+                    MetricValue_ = workToBeDone
+                });
+            }
+            catch (Exception ex)
+            {
+                logger.Log(ex.Message);
+            }
+            return resp;
+        }
+
+        private CosmosDbTrigger CreateTriggerFromMetadata(MapField<string, string> scalerMetadata)
+        {
+            var trigger = new CosmosDbTrigger()
+            {
+                CollectionName = scalerMetadata[CosmosDbTriggerMetadata.CollectionName],
+                DatabaseName = scalerMetadata[CosmosDbTriggerMetadata.DatabaseName],
+                DocDBConnectionString = scalerMetadata[CosmosDbTriggerMetadata.DocDBConnectionString],
+                LeasesDocDBConnectionString = scalerMetadata[CosmosDbTriggerMetadata.LeasesDocDBConnectionString],
+                LeaseDatabaseName = scalerMetadata[CosmosDbTriggerMetadata.LeaseDatabaseName],
+                LeaseCollectionName = scalerMetadata[CosmosDbTriggerMetadata.LeaseCollectionName],
+                LeaseCollectionPrefix = string.Empty,
+                AccountName = string.Empty,
+            };
+
+            // Optional values
+            if (scalerMetadata.TryGetValue(CosmosDbTriggerMetadata.LeaseCollectionPrefix, out string leasePrefix))
+            {
+                trigger.LeaseCollectionPrefix = leasePrefix;
+            }
+
+            if (scalerMetadata.TryGetValue(CosmosDbTriggerMetadata.AccountName, out string accountName))
+            {
+                trigger.AccountName = accountName;
+            }
+
+            return trigger;
+        }
+
+        private async Task<long> GetEstimatedWork (CosmosDbTrigger trigger)
+        {
+            var host = CreateHost(trigger, out string warningMessage);
+            long workToBeDone = 0;
+
+            if (host != null)
+            {
+                workToBeDone = await host.GetEstimatedRemainingWork();
+            }
+            else
+            {
+                logger.Log(warningMessage);
+            }
+
+            return workToBeDone;
+        }
+
+#pragma warning disable CS0618 // Type or member is obsolete

```

Comment:
We shouldn't add obsolete things for new scalers, can you migrate to the new version please?